jib-cli/src/test/java/com/google/cloud/tools/jib/ArtifactProcessorsTest.java

```diff
@@ -174,4 +185,50 @@ public void testDetermineJavaMajorVersion_emptyClassFile() throws URISyntaxExcep
             () -> ArtifactProcessors.determineJavaMajorVersion(jarPath));
     assertThat(exception).hasMessageThat().startsWith("Reached end of class file (class1.class)");
   }
+
+  @Test
+  public void testFromWar_noJettyBaseImageAndNoAppRoot() {
+    when(mockCommonContainerConfigCliOptions.getFrom()).thenReturn(Optional.of("base-image"));
+
+    IllegalStateException exception =
+        assertThrows(
+            IllegalStateException.class,
+            () ->
+                ArtifactProcessors.fromWar(
+                    Paths.get("my-app.war"),
+                    mockCacheDirectories,
+                    mockWarCommand,
+                    mockCommonContainerConfigCliOptions));
+
+    assertThat(exception)
+        .hasMessageThat()
+        .startsWith("Please set the app root of the container with `--app-root`");
+  }
+
+  @Test
+  public void testFromWar_noJettyBaseImageAndAppRootPresent_success() {
+    when(mockCommonContainerConfigCliOptions.getFrom()).thenReturn(Optional.of("base-image"));
+    when(mockWarCommand.getAppRoot()).thenReturn(Optional.of(AbsoluteUnixPath.get("/app-root")));
+    when(mockCacheDirectories.getExplodedJarDirectory()).thenReturn(Paths.get("exploded-jar"));
+    ArtifactProcessor processor =
+        ArtifactProcessors.fromWar(
+            Paths.get("my-app.war"),
+            mockCacheDirectories,
+            mockWarCommand,
+            mockCommonContainerConfigCliOptions);
+
+    assertThat(processor).isInstanceOf(StandardWarExplodedProcessor.class);
+  }
+
+  @Test
+  public void testFromWar_jettyBaseImageSpecified_success() {
+    when(mockCommonContainerConfigCliOptions.getFrom()).thenReturn(Optional.of("jetty"));
```

Comment:
How about "jetty:tag" to make it more interesting?

Reply:
Good idea.